src/org.ts

```diff
@@ -799,43 +799,42 @@ export class Org extends AsyncCreatable<Org.Options> {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   private async removeUsers(throwWhenRemoveFails: boolean): Promise<void> {
     this.logger.debug(`Removing users associate with org: ${this.getOrgId()}`);
-    const config = await this.retrieveOrgUsersConfig();
+    const [config, authInfos, aliases] = await Promise.all([
+      this.retrieveOrgUsersConfig(),
+      this.readUserAuthFiles(),
+      Aliases.create(Aliases.getDefaultOptions()),
+    ]);
     this.logger.debug(`using path for org users: ${config.getPath()}`);
-    if (await config.exists()) {
```

Comment:
why was `config.exists()` returning (I'm assuming) `false` when the config actually existed?

Reply:
Because it's based on <orgId>.json (true for sandboxes) but not true for scratch orgs, which use <username>.json.

There's several layers of `remove` that are redundant (see some of the comments) as it tries to handle both scratch org and sandboxes and non-deletable orgs that you can still delete auths/configs/aliases for.

Some of my wilder ideas were a single, idempotent "ensureRemoved" method that looks in all the places for all the things and removes anything it finds but handles not finding things.

Reply:
```ts
 public async retrieveOrgUsersConfig(): Promise<OrgUsersConfig> {
    return await OrgUsersConfig.create(OrgUsersConfig.getOptions(this.getOrgId()));
}
```

In the scenario for this bug, there is no OrgUsers file (the one scratch orgs have with the orgId only after you've run `user:create`) so the alias and defaultusername removal gets skipped